python/cugraph/structure/graph.py

```diff
@@ -565,945 +277,394 @@ def from_numpy_matrix(self, np_matrix):
-            This cudf.Series wraps a gdf_column of size V + 1 (V: number of
-            vertices).
-            The gdf column contains the offsets for the vertices in this graph.
-            Offsets are in the range [0, E] (E: number of edges).
-        index_col : cudf.Series
-            This cudf.Series wraps a gdf_column of size E (E: number of edges).
-            The gdf column contains the destination index for each edge.
-            Destination indices are in the range [0, V) (V: number of
-            vertices).
-        value_col : cudf.Series or ``None``
-            This pointer is ``None`` for unweighted graphs.
-            For weighted graphs, this cudf.Series wraps a gdf_column of size E
-            (E: number of edges).
-            The gdf column contains the weight value for each edge.
-            The expected type of the gdf_column element is floating point
-            number.
-
+        ---------
+        df : cudf.DataFrame or dask_cudf.DataFrame
+            Original DataFrame with new column containing internal vertex
+            id
         """
-        if self.distributed:
-            raise Exception("Not supported for distributed graph")
-        if self.transposedadjlist is None:
-            if self.adjlist is not None and type(self) is Graph:
-                off, ind, vals = (
-                    self.adjlist.offsets,
-                    self.adjlist.indices,
-                    self.adjlist.weights,
-                )
-            else:
-                (
-                    off,
-                    ind,
-                    vals,
-                ) = graph_primtypes_wrapper.view_transposed_adj_list(self)
-            self.transposedadjlist = self.transposedAdjList(off, ind, vals)
-
-            if self.batch_enabled:
-                self._replicate_transposed_adjlist()
-
-        return (
-            self.transposedadjlist.offsets,
-            self.transposedadjlist.indices,
-            self.transposedadjlist.weights,
+        return self.renumber_map.add_internal_vertex_id(
+            df,
+            internal_column_name,
+            external_column_name,
+            drop,
+            preserve_order,
         )
 
-    def delete_adj_list(self):
-        """
-        Delete the adjacency list.
+    def clear(self):
         """
-        self.adjlist = None
-
-    def get_two_hop_neighbors(self):
+        Empty the graph.
         """
-        Compute vertex pairs that are two hops apart. The resulting pairs are
-        sorted before returning.
+        self.Impl = None
 
-        Returns
-        -------
-        df : cudf.DataFrame
-            df[first] : cudf.Series
-                the first vertex id of a pair, if an external vertex id
-                is defined by only one column
-            df[second] : cudf.Series
-                the second vertex id of a pair, if an external vertex id
-                is defined by only one column
+    def is_bipartite(self):
```

Comment:
Updated to cugraph functionality, i.e. it can also be used as cugraph.is_bipartite(g)
This function in future releases can be updated to call coloring algorithm to inspect and evaluate if graph is bipartite